candidate_base.go

```diff
@@ -10,6 +10,7 @@ import (
 	"hash/crc32"
 	"io"
 	"net"
+	"reflect"
```

Comment:
Just started reading, but would prefer to never use reflection.

Makes things harder on tinygo and we lose safety